File: test/end-to-end/PointClass/C#/NamedPoint.cs
//
using PointClass;

namespace PointClass
{

    public class NamedPoint : Point
    {
        private string name;

        public NamedPoint(int x, int y, string name)
            : base(x, y)
        {
            this.name = name;
        }

        public string GetLabel()
        {
            return string.Format("{0} at [{1}, {2}] with vector {3}", this.name, this.X, this.Y, this.Vector);
        }
    }
}
//

File: test/end-to-end/PointClass/TypeScript/NamedPoint.ts
import { Point } from "./Point";

export class NamedPoint extends Point {
    private name: string;

    public constructor(x: number, y: number, name: string) {
        super(x, y);
        this.name = name;
    }

    public getLabel(): string {
        return `${this.name} at [${this.x}, ${this.y}] with vector ${this.vector}`;
    }
}
